packages/snap-tracker/src/types.ts

```diff
@@ -214,20 +183,19 @@ export type PreflightRequestModel = {
 };
 
 export interface TrackMethods {
-	event: (payload: BeaconPayload) => BeaconEvent | undefined;
-	error: (data: TrackErrorEvent) => BeaconEvent | undefined;
+	error: (data: TrackErrorEvent) => undefined;
 	shopper: {
-		login: (data: ShopperLoginEvent, siteId?: string) => BeaconEvent | undefined;
+		login: (data: ShopperLoginEvent, siteId?: string) => undefined;
 	};
 	product: {
-		view: (data: ProductViewEvent, siteId?: string) => BeaconEvent | undefined;
+		view: (data: Item | ProductViewEvent, siteId?: string) => undefined;
 		click: (data: ProductClickEvent, siteId?: string) => BeaconEvent | undefined;
 	};
 	cart: {
-		view: (data: CartViewEvent, siteId?: string) => BeaconEvent | undefined;
+		view: (data: CartViewEvent | CartSchemaData, siteId?: string) => undefined;
 	};
 	order: {
-		transaction: (data: OrderTransactionData, siteId?: string) => BeaconEvent | undefined;
+		transaction: (data: OrderTransactionData | OrderTransactionSchemaData, siteId?: string) => undefined;
```

Comment:
Should we remove `Item` usage here? Also, I don't think we need to have these methods support the new data schema types either... do we?